URBNAlert/Classes/AlertView.swift

```diff
@@ -209,12 +209,12 @@ extension AlertView: AlertButtonContainer {
 
 extension AlertView: UITextFieldDelegate {
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
-        let charCount = textField.text?.count ?? 0
-        if let charCount = textField.text?.count, range.length + range.location > charCount {
+        let charCount = textField.text?.utf16.count ?? 0
+        if let charCount = textField.text?.utf16.count, range.length + range.location > charCount {
             return false
         }
         
-        let newLength = charCount + string.count - range.length
+        let newLength = charCount + string.utf16.count - range.length
```

Comment:
Ok I did a dive into what's going on here.. 

- When entering an emoji, `NSRange`'s length reads as `2`, as its counting "UTF-16 code-units", or the length in memory it takes up
- Swifts `String.count` returns the straight up character count, for a single emoji would simply be `1`, which is why you were unable to delete an Emoji once entered. It would end up checking `2 > 1` which would return false, meaning whatever the user just pressed would be ignored (even a backspace)
- Calling `.utf16.count` also returns `2` for a single emoji though, which is why this is working here as its matching what `NSRange` is counting. While this "fixes" the issue, I think we can better yet modernize and clean the code up a bit. I also found an existing bug along the way.

Bug Found: This code is in place purely so we can enforce a `maxLength` for our alert textfields. Right now if you were set the `maxLength` for the promo alert to `3`, you are only ever allowed to enter 2 characters-- we are doing some bad math. More on that below in the snippet

To modernize we should start using Swift's `Range` rather than `NSRange`. From what I've read, it will better match what `String.count` is doing. It would be cleaner if we could just use `String.count` and not have to deal with the range values at all-- as `.count` will return the actual number of characters, regardless if its a letter, emoji, etc.

I was playing around and came up with some code that seems to fix both the emoji issue, and the maxLength issue:
```
func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
        if let oldString = textField.text, let swiftRange = Range(range, in: oldString) {
            let newString = oldString.replacingCharacters(in: swiftRange, with: string)
            
            return newString.count <= configuration.styler.textField.maxLength
        }
        
        return false
    }
```

What this snippet does is grabs the "old" string, creates a `Range` out of the `NSRange`,  then creates a "new" string that appends the character the user just typed with the "old" string. Regardless if emojis, or text is entered we will get the correct count of individual characters, which is what we want.

This was a fun one 🙂 Heres some references I came across researching this:

https://stackoverflow.com/a/45506737/480415
https://stackoverflow.com/a/24037759/480415
This ones a little older/outdated but still gives some good insight to the inner workings going on
https://arasthel.com/a-tale-of-two-string-lengths-and-emojis/

@codecaffeine please correct me if I am wrong at all here-- I know you've taken some deep lives into String's before

Reply:
This is all great stuff! Lets talk about it in MTM